### DISCUSSION TITLE
New feature: rounded corners + shadow

### DISCUSSION THREAD

**Discussion by nvtoroy** [flameshot-org/flameshot#3159] [Ideas]
Hello.
I think it would be great to have a feature that would make the corners of the window being screenshoted rounded to a given radius, as well as add a small shadow under that window.
Screenshots would get a neat modern shape and volume.
The screenshot below shows an example of what I would like a screenshot to look like automatically after selecting a field and saving it to a file or memory.
![2023-04-10_05-39](https://user-images.githubusercontent.com/90771587/230814661-31dcf0da-7481-4816-88df-3c3c84d0caa3.png)

**Comment by mmahmoudian** [comment]
Both are already suggested before. Searching through open and closed issues can usually save your time

**Reply by lriffel-cinchhs** [reply]
Can you point us to the open Ideas containing this Idea?  I was unable to find anything else with the words "rounded" or "shadow".

**Reply by mmahmoudian** [reply]
Of course. For shadow:
- #1919
- #1249
- #958

For rounded corners:
- #1156

For border:
- #690

**Reply by mmahmoudian** [reply]
if you need to do it now, I have put together some simple scripts that you can modify to your liking:

https://github.com/flameshot-org/flameshot/issues/690#issuecomment-829972312

But this only works on Linux and with minor modification on macOS.

**Reply by lriffel-cinchhs** [reply]
Thank you, I see why I couldn't find them.   I was looking in the Discussions section under Ideas where they can be up-voted rather than the issues section.  

The border is the part I would like to do, but I'm on Windows.

**Reply by mmahmoudian** [reply]
Yes, Github is a little bit unintuitive, especially the mix use of discussions and issues. In this project we use discussions literally for discussions. Not for feature request and not for bug reports.

Regarding the border, I'm sorry that I can't help. I stopped using Windows over a decade ago (Windows 7) and I have been enjoying my life ever since. I guess honestly you have three choices if you want auto-bordering:
1. Use Linux (it is off topic but I will be glad to help)
2. Wait until the feature lands in Flameshot
3. Use another screenshot tool (e.g Greenshot which is also OpenSource)

Of course as a maintainer of Flameshot I would like the order above, but the fact is switching to another very good screenshot tool is the easiest I guess. I don't know if Greenshot has this feature, but I think that is your best bet.

**Reply by lriffel-cinchhs** [reply]
Thanks for the reply.

1. I use Linux extensively, but need to screenshot in Windows much more often.
2. Waiting
3. Used Greenshot in the past, but like Flameshot enough to use it over Greenshot without the border feature. (I also don't know if it can do borders).

I switched from SnagIt to Flameshot after Snagit started slowing down as more screen shots were taken and eventually had to be re-installed to get the speed back.  The only two features I miss are the automatic border and the quick list of recent screen shots.

**Reply by mmahmoudian** [reply]
In Flameshot for the sake of privacy we do not record anything other than:
- the imgur uploads history (because they are uploaded anonymously and users tend to want to delete them later and need the delete link)
- the changes in the settings

You can also [create a feature request](https://github.com/flameshot-org/flameshot/issues/new?assignees=&labels=Enhancement&projects=&template=feature-request.yaml) and propose the idea of keeping some sort of ledger for screenshots. I'm sure it would be easy to write a simple [plugin](https://github.com/flameshot-org/flameshot/issues?q=is%3Aopen+is%3Aissue+label%3APlugin) for this when the plugin system (#2529) is implemented.

In the meantime, it is easy and quick in bash/zsh to put together some sort of history management system (txt, csv, or sqlite), and use fzf or rofi as the interactive UI. But again, on Windows I have no idea how easily bash/zsh can be ported to PowerShell. I might write a helper script for that in bash at somepoint, and I will ping you if/when it was ready.

**Comment by nvtoroy** [comment]
sorry

**Reply by mmahmoudian** [reply]
Well, as I said, searching would have saved **your time**, therefore, there was no need to say sorry 😊

If you have other ideas, please feel free to suggest them (of course after searching). Thanks.
